ddsrecorder/src/cpp/main.cpp

```diff
@@ -0,0 +1,544 @@
+                            }
+                            if (prev == CommandCode::PAUSE)
+                            {
+                                receiver.publish_status(CommandCode::START, CommandCode::PAUSE);
+                            }
+                            break;
+
+                        case CommandCode::PAUSE:
+                            if (!first_iter)
+                            {
+                                mcap_handler->pause();
+                            }
+                            if (prev == CommandCode::START)
+                            {
+                                receiver.publish_status(CommandCode::PAUSE, CommandCode::START);
+                            }
+                            break;
+
+                        case CommandCode::EVENT:
+                            mcap_handler->trigger_event();
+                            // TODO: transition to STOP/START/CLOSE if argument non-empty (command=x + continue)
+                            break;
+
+                        case CommandCode::STOP:
+                        case CommandCode::CLOSE:
+                        case CommandCode::NONE:
+                            // Unreachable
+                            logError(DDSRECORDER_EXECUTION,
+                                    "Reached an unstable execution state: command " << command << " case.");
+                            continue;
+
+                        default:
+                        case CommandCode::UNKNOWN:
+                            break;
+                    }
+                    receiver.wait_for_command();
+                    prev = command;
+                    command = receiver.command_received();
+                    first_iter = false;
+
+                } while (command != CommandCode::STOP && command != CommandCode::CLOSE && command != CommandCode::NONE);
+            } while (command != CommandCode::CLOSE && command != CommandCode::NONE);
+
+            // Transition to CLOSED state
+            receiver.publish_status(CommandCode::CLOSE, prev);
+        }
+        else
+        {
+            // Start recording right away
+            auto recorder = create_recorder(configuration);
+
+            // Create File Watcher Handler
+            std::unique_ptr<eprosima::utils::event::FileWatcherHandler> file_watcher_handler;
+            if (file_path != "")
+            {
+                file_watcher_handler = create_filewatcher(recorder, file_path);
+            }
+
+            // Create Periodic Handler
+            std::unique_ptr<eprosima::utils::event::PeriodicEventHandler> periodic_handler;
+            if (reload_time > 0 && file_path != "")
+            {
+                periodic_handler = create_periodic_handler(recorder, file_path, reload_time);
+            }
+
+            // Wait until signal arrives
+            close_handler.wait_for_event();
+        }
+
+        logUser(DDSRECORDER_EXECUTION, "Stopping DDS Recorder.");
```

Comment:
this message should not be here

Reply:
Why not? Maybe it should say closing instead?

Reply:
Because recorder has already being stopped. It had sense before closing it.
However is a NIT comment